app/modules/booking/helpers.ts

```diff
@@ -0,0 +1,29 @@
+import { isAfter, isSameDay } from "date-fns";
+
+/**
+ * This function checks if the booking is being early checkout
+ */
+export function isBookingEarlyCheckout(from: string | Date): boolean {
+  const now = new Date();
+
+  /** If the booking is for today, it's not early checkout */
+  if (isSameDay(from, now)) {
+    return false;
+  }
+
+  return isAfter(from, now);
+}
+
+/**
+ * This function checks if the booking is being early checkin
+ */
+export function isBookingEarlyCheckin(to: string | Date) {
```

Comment:
This only checks date. Should also check time and even if its 5 minutes earlier it should adjust it.